api/src/transactions.rs

```diff
@@ -154,6 +200,18 @@ async fn handle_submit_bcs_transactions(
     Ok(Transactions::new(context)?.create(txn).await?)
 }
 
+async fn handle_simulate_bcs_transactions(
+    body: bytes::Bytes,
+    context: Context,
+) -> Result<impl Reply, Rejection> {
+    fail_point("endpoint_simulate_bcs_transactions")?;
+    let txn = bcs::from_bytes(&body)
+        .map_err(|err| Error::invalid_request_body(format!("deserialize error: {}", err)))?;
+    Ok(Transactions::new(context)?
+        .simulate_from_request(txn)
```

Comment:
shouldn't this be calling simulate?

Reply:
Right. Bad copy pasta and no test case for this.

Reply:
I removed the BCS case entirely because I don't have time to test it..